File: apps/web/src/schemas/pagination.ts
export type PageRequest = {
  page: number;
  size: number;
};

export type Page<T> = {
  items: T[];
  total: number;
  totalPages: number;
  size: number;
};

export const emptyPage = <T>(): Page<T> => ({
  items: [],
  total: 0,
  totalPages: 1,
  size: 0,
});

File: apps/auth/src/models/pagination.rs
use serde::{Deserialize, Serialize};
use utoipa::ToSchema;

#[derive(Debug, Serialize, Deserialize, ToSchema)]
pub struct PaginatedRequest {
    pub page: Option<u64>,
    pub size: Option<u64>,
}

#[derive(Debug, Serialize, Deserialize, ToSchema)]
#[serde(rename_all = "camelCase")]
pub struct PaginatedResponse<T> where T: Serialize {
    pub total: u64,
    pub page: u64,
    pub total_pages: u64,
    pub items: Vec<T>,
}
